process_testcase.go

```diff
@@ -574,13 +576,43 @@ func processVariableAssignments(ctx context.Context, tcName string, tcVars H, ra
 
 	for varname, assignment := range stepAssignment.Assignments {
 		Debug(ctx, "Processing %s assignment", varname)
+		lazilyLoadJSON := os.Getenv(LAZY_JSON_EXPANSION_FLAG) == FLAG_ENABLED
+		if lazilyLoadJSON {
+			if strings.Contains(assignment.From, "json") {
```

Comment:
Why do you need to check the `json` parts in the From attribute? It's possible to have a json variable without a name containing `json` string.

Reply:
we want to check the assignments for the keys that we are generating from the unmarshalling. For example we are making new keys with the name `result.systemoutjson` for the key `result.systemout` the check here is to check if the  key has that `json` substring in order to check if we need to unmarshal that payload.